File: Example/MXParallaxHeader/UIScrollView/UIScrollViewExample.m
//
//  MXViewController.m
//  MXParallaxHeader
//
//  Created by Maxime Epain on 10/28/2015.
//  Copyright (c) 2015 Maxime Epain. All rights reserved.
//

#import "UIScrollViewExample.h"
#import <MXParallaxHeader/MXParallaxHeader.h>

@interface UIScrollViewExample ()
@end

@implementation UIScrollViewExample

- (void)viewDidLoad
{
    [super viewDidLoad];
	
    // Parallax Header
    UIImageView *header = [UIImageView new];
    header.image = [UIImage imageNamed:@"success-baby"];
    header.contentMode = UIViewContentModeScaleAspectFill;
    
    self.tableView.parallaxHeader.view = header;
    self.tableView.parallaxHeader.height = 150;
    self.tableView.parallaxHeader.mode = MXParallaxHeaderModeFill;
    self.tableView.parallaxHeader.minimumHeight = 20;
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

#pragma mark <UITableViewDelegate>

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    self.tableView.parallaxHeader.height = indexPath.row * 10;
}

#pragma mark <UITableViewDataSource>

- (NSInteger) tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return 50;
}

-(UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"Cell"];
    cell.textLabel.text = [NSString stringWithFormat:@"Height %ld", (long)indexPath.row * 10];
    return cell;
}

@end

File: Example-Swift/MXParallaxHeader-Swift/UIScrollViewExample.swift
// UIScrollViewExample.swift
//
// Copyright (c) 2015 Maxime Epain
//
// Permission is hereby granted, free of charge, to any person obtaining a copy
// of this software and associated documentation files (the "Software"), to deal
// in the Software without restriction, including without limitation the rights
// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
// copies of the Software, and to permit persons to whom the Software is
// furnished to do so, subject to the following conditions:
//
// The above copyright notice and this permission notice shall be included in
// all copies or substantial portions of the Software.
//
// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
// THE SOFTWARE.

import UIKit
import MXParallaxHeader

class UIScrollViewExample: UITableViewController {

    override func viewDidLoad() {
        super.viewDidLoad()

        // Parallax Header
        let header = UIImageView()
        header.image = UIImage(named:"success-baby")
        header.contentMode = UIViewContentMode.ScaleAspectFill
        
        self.tableView.parallaxHeader.view = header
        self.tableView.parallaxHeader.height = 150
        self.tableView.parallaxHeader.mode = MXParallaxHeaderMode.Fill
        self.tableView.parallaxHeader.minimumHeight = 20
    }

    // MARK: - Table view data source

    override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return 50
    }

    
    override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCellWithIdentifier("Cell", forIndexPath: indexPath)
        cell.textLabel!.text = String(format: "Height %ld", indexPath.row * 10)
        return cell
    }
    
    override func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
        self.tableView.parallaxHeader.height = CGFloat(indexPath.row * 10)
    }

}
